plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
On Tue, May 17, 2016 at 04:09:24PM -0700, Michael Grosser wrote:
> >      def deployment_labels
> > -      # Deployment labels should not include the ids of the release, role or deploy groups
> > -      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
> > +      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
> 
> we use a different namespace for every environment, so role and project are unique enough ... and it allows us to not have to change any labels around and still be able to deploy + use local kubectl create too ... works for you too ?

Yes, role and project should be enough. Not sure what is the problem of using
the deploy_group too in the slice. It's just for sure unique  (doesn't depend on
the configuration of the cluster) and doesn't cause any problems with updating
labels on deploy. Right?